### PR TITLE
Add bid adapter for Missena

### PR DESCRIPTION
**Pull request by dev-missena**
## Type of change

- [x] New bidder adapter

## Description of change

Test parameters for validating bids:

```
  {
    bidder: "missena",
    params: {
      apiKey: "PA-34745704",
    },
  }
```

### Contact

- jney@missena.com
- [x] official adapter submission

## Documentation PR

https://github.com/prebid/prebid.github.io/pull/2679

## Other information

This adapter doesn't support SafeFrame.

### CODE REVIEW

File: modules/missenaBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import * as utils from '../src/utils.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+
+const BIDDER_CODE = 'missena';
+const ENDPOINT_URL = 'https://bid.missena.io/';
+
+export const spec = {
```
Comment (blingster7) [inline modules/missenaBidAdapter.js:8]: can you please add your gvlid if you have one?

Comment (dev-missena) [reply modules/missenaBidAdapter.js:8]: Thanks! I've added our global vendor ID to the spec.

**Comment by blingster7** [comment]
Can you please also add your bidAdapter to prebid.docs.io? 

https://docs.prebid.org/dev-docs/bidder-adaptor.html

https://github.com/prebid/prebid.github.io

File: modules/missenaBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return typeof bid == 'object' && !!bid.params.apiKey;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
```
[the remark this answers is not in the document]
Comment (blingster7) [reply modules/missenaBidAdapter.js:30]: ah wait you're not passing any imp objects in yet? this is just for testing right?

File: modules/missenaBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const req = validBidRequests[0];
+    const apiKey = req.params.apiKey;
+
```
Comment (blingster7) [inline modules/missenaBidAdapter.js:33]: Being picky here but shouldn't `req` and `api` key be evaluated for each valid bid in map below? Assuming there are multiple missena bids in a request...

**Comment by blingster7** [comment]
Can you merge with master again to see if it makes the CircleCI tests pass?

File: modules/missenaBidAdapter.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dev-missena) [reply modules/missenaBidAdapter.js:30]: The response from Missena doesn't rely on the banner size.
So we don't transfer it on purpose. Hope this clarify!

File: modules/missenaBidAdapter.js (hunk 3 above)
Comment (dev-missena) [reply modules/missenaBidAdapter.js]: Thanks for catching it!

**Comment by dev-missena** [comment]
Thanks for the merge with master. I've also added the referer info to the request payload.
Looks like CircleCI is failing again but on non-Missena tests.

**Comment by cedricss** [comment]
Merged with master. Tests are passing 👌

**Comment by dev-missena** [comment]
Merged with master. Tests are passing 👌

**Review by blingster7 — APPROVED** [review]

**Pull request merged**